custom derives don't work on enums

<!-- Thanks for filing a bindgen issue! We appreciate it :-) -->

@trobanga pointed out that PR #2059 (implementing #1089) doesn't seem to work if we try to add a custom derive to an `enum` instead of a struct.

I can see the problem if I try to extend the unit test from #2059 like this:

### Input C/C++ Header

```C++
enum my_rustified_enum {
  ELEVEN = 11,
  TWELVE,
  THIRTEEN,
};
```

```rust
#[test]
fn test_custom_derive() {
    use std::cmp::{PartialEq, Ordering};

    let test3 = unsafe { bindings::my_rustified_enum::ELEVEN };
    let test4 = unsafe { bindings::my_rustified_enum::TWELVE };

    assert!(test3 < test4);
}
```

### Bindgen Invocation

<!-- Place either the `bindgen::Builder` or the command line flags used here. -->

I tested this using `cargo test` within `bindgen-integration/`, with the custom derives specified like this:
```rust
fn add_derives(&self, name: &str) -> Vec<String> {
    if name == "my_rustified_enum" {
        vec!["PartialOrd".into()]
    } else {
        vec![]
    }
}
```

### Actual Results

``` txt
$ cargo test
   Compiling bindgen-integration v0.1.0 (.../rust-bindgen/bindgen-integration)
error[E0369]: binary operation `<` cannot be applied to type `root::my_rustified_enum`
   --> src/lib.rs:289:19
    |
289 |     assert!(test3 < test4);
    |             ----- ^ ----- root::my_rustified_enum
    |             |
    |             root::my_rustified_enum
    |
    = note: an implementation of `std::cmp::PartialOrd` might be missing for `root::my_rustified_enum`
```

### Expected Results

`PartialOrd` should have been added to the `#[derive(...)` attribute, allowing this test to compile.

--- a/bindgen-integration/build.rs
+++ b/bindgen-integration/build.rs
@@ -126,6 +126,10 @@ impl ParseCallbacks for MacroCallback {
             vec![
                 "PartialEq".into(),
             ]
+        } else if name == "MyOrderedEnum" {
+            vec![
+                "PartialOrd".into(),
+            ]
         } else {
             vec![]
         }

--- a/bindgen-integration/cpp/Test.h
+++ b/bindgen-integration/cpp/Test.h
@@ -234,3 +234,10 @@ typedef union {
 } Coord;
 
 Coord coord(double x, double y, double z, double t);
+
+// Used to test custom derives on enum. See `test_custom_derive`.
+enum MyOrderedEnum {
+  MICRON,
+  METER,
+  LIGHTYEAR,
+};

--- a/bindgen-integration/src/lib.rs
+++ b/bindgen-integration/src/lib.rs
@@ -275,4 +275,14 @@ fn test_custom_derive() {
     let test1 = unsafe { bindings::Test::new(5) };
     let test2 = unsafe { bindings::Test::new(6) };
     assert_ne!(test1, test2);
+
+    // The `add_derives` callback should have added `#[derive(PartialOrd)]`
+    // to the `MyOrderedEnum` enum. If it didn't, this will fail to compile.
+
+    let micron = unsafe { bindings::MyOrderedEnum::MICRON };
+    let meter = unsafe { bindings::MyOrderedEnum::METER };
+    let lightyear = unsafe { bindings::MyOrderedEnum::LIGHTYEAR };
+
+    assert!(meter < lightyear);
+    assert!(meter > micron);
 }

--- a/src/codegen/mod.rs
+++ b/src/codegen/mod.rs
@@ -3014,6 +3014,16 @@ impl CodeGenerator for Enum {
                     derives.push(derive);
                 }
             }
+
+            // The custom derives callback may return a list of derive attributes;
+            // add them to the end of the list.
+            let custom_derives;
+            if let Some(cb) = &ctx.options().parse_callbacks {
+                custom_derives = cb.add_derives(&name);
+                // In most cases this will be a no-op, since custom_derives will be empty.
+                derives.extend(custom_derives.iter().map(|s| s.as_str()));
+            };
+
             attrs.push(attributes::derives(&derives));
         }
 

